hyperspy/_signals/spectrum.py

```diff
@@ -84,7 +84,10 @@ def _spikes_diagnosis(self, signal_mask=None,
             dc = dc[~navigation_mask, :]
         der = np.abs(np.diff(dc, 1, -1))
         plt.figure()
-        plt.hist(np.ravel(der.max(-1)), 100)
+        if len(dc.shape) == 1:
```

Comment:
That's a good point... Not really! I added some:
![spike1](https://cloud.githubusercontent.com/assets/1278301/8987085/3251e4e0-36ad-11e5-80ba-b0f195520e9f.png)
![spike2](https://cloud.githubusercontent.com/assets/1278301/8987086/3252f006-36ad-11e5-8e25-57a74db56a68.png)

Tests incoming...